rasterio/session.py

```diff
@@ -97,11 +98,14 @@ def cls_from_path(path):
 
         path = _parse_path(path)
 
+        aws_path_regex = r"^s3://|^https?://.*\.s3\.amazonaws\.com/"
+        aws_path_regex = os.getenv("AWS_PATH_REGEX", aws_path_regex)
+
         if isinstance(path, _UnparsedPath) or path.is_local:
             return DummySession
 
         elif (
-            path.scheme == "s3" or "amazonaws.com" in path.path
```

Comment:
I think keeping `path.scheme == "s3" or` should be safe and makes sense to keep.

Reply:
rolled back